instructor-training-process.md

```diff
@@ -0,0 +1,154 @@
+1.  **Welcoming to the community.**
+    1.  Once the trainee has passed the demonstration lesson,
+        we generate a PDF certificate for the trainee by running a command-line script in
+        [the certification repository][cert-repo]
+	and send it to her along with instructions for the items below.
+    1.  The instructions sent to the new instructor tell her to:
+        1.  join the `discuss` and `instructors` lists
+	1.  fill in a form on AMY to tell us where she is, what she's comfortable teaching, etc., and
+	1.  send us a short biography and photo to add to [our website][team-page].
+    1.  We manually add the biography and photo to the website
+        (since they almost always require reformatting).
+
+## Resources
+
+1.  [AMY][amy] is the web application we use to track instructors and workshops.
+    Most of it is only accessible to administrators,
+    but it does have a form where instructors can supply or update information about themselves.
+1.  There's a web form on AMY that people can use to apply for instructor training
+    (currently disabled).
+1.  We keep a list of trainees in a Google Docs spreadsheet with the following fields,
+    which we upate manually:
+    1.  name
+    1.  contact email
+    1.  which training course they took part in
+    1.  the name of the Software Carpentry lesson they're concentrating on (if any)
+    1.  the name of the Data Carpentry lesson they're concentrating on (if any)
+    1.  the URL of their Software Carpentry exercise submission (if any)
+    1.  whether or not they have submitted something for Data Carpentry
+    1.  the date and moderator of the discussion session they have taken part in
+        *   we currently store both values in one column
+	*   we do not record both a Software Carpentry and a Data Carpentry discussion session (though we should)
+    1.  the date on which they passed their Software Carpentry lesson demo (if any)
+	*   this date becomes the award date of the PDF certificate
+        *   we do not currently record the name of the person who was their instructor, or the name of the person who did their final test
+    1.  the date on which they passed their Data Carpentry lesson demo (if any)
+        *   the same caveats apply
+1.  Mentors offer lesson discussion sessions,
+    and trainees sign up for them,
+    on an Etherpad.
+1.  Trainers offer checkout sessions on a second Etherpad.
+    Trainees sign up for sessions on that Etherpad,
+    which we also presently use for taking notes during their sessions.
+1.  Discussion and checkout sessions are marked in Software Carpentry's community calendar
+    (a Google calendar).
+
+## Issues
+
+1.  As of Jan 2016,
+    the form for individuals to apply to take part in training has been disabled
+    while we try to clear our backlog (and come up with a better process).
+1.  We used a Google web form and a spreadsheet for group applications and sorted manually.
```

Comment:
On Wed, Mar 09, 2016 at 09:13:17AM -0800, chrismedrela wrote:
> instructors with a badge within D miles of [airport code] who have
> taught < N workshops in the last M months who are comfortable
> teaching topic X

This isn't too hard to write out in a GitHub- (or Xapian- [1], etc.)
-style searches.  Probably something like:

  badge:swc-instructor airport:yvr radius:20 workshops-lt:2 workshop-start-gt:2015-10-01

If you're unpacking it to Django or SQL, each of those parameters maps
cleanly onto an orthogonal constraint (and yeah, workshops-lt would
require aggregation).  So it would be a fair amount of work to
implement, but it doesn't seem unmanageable.  And it certainly seems
like less work than sorting manually every time you need a particular
query ;).

[1]: http://getting-started-with-xapian.readthedocs.org/en/latest/concepts/search/queryparser.html#searching-with-prefixes